### ISSUE TITLE
Auto update Apk on tablet options

### ISSUE THREAD

**Issue by leomaxi** [open-learning-exchange/take-home#23]
_edited: by Kyle Mathias 4-28-17_
We are trying to look into different ways to Automatically update the application. Right now, all we have is sending a new APK by email, then having someone install it (usually a tech genius). There couple of different ideas we had to get updates to the user. They are in order of preference.

- [ ] Update through the BeLLApp
* **How**: When the Android device sync with the nation, it sees that it is out of date, and goes and pulls the update from a Resource module in BeLLApp/CouchDB
* **Basic Flow**: Device connects to raspberry pi->user logs in->gets notified of update->clicks update->App pulls update from BeLLApp Resurce/CouchDB and it installs automatically.
* **Potential Problem** The app is 600mb. Pull a 600mb module from the nation will hang/download for a long time. Possible solution would be to only pull different file changes OR have it get downloaded out side of the App.

- [ ] Update through google play.
* **How**: User connects to the internet 
* **Basic Flow**: User Connects to an internet source->Downloads update from app store->Installs Automatically.
* **Potential Problem**: Most users do not have access to the internet.

We would like everyone to look into possible ways to solve this problem through both options. We are going to meet in a week from today to discus the solutions we each found and how to go about starting implementation. We will meet Friday, the 5th of May at 10am PST/ 12pm CST / 1pm EST.

**Comment by kylemathias** [comment]
Added some context.

**Comment by lmmrssa** [comment]
@leonardmensah  You mentioned its going to be 600mb, Is it just apk or you mean the source code? APK should not be such heavy. But if you are looking for data of app then it could be. For that if we keep the apk name same then updating apk its possible to maintain the old data and just sync the design/views.

**Comment by kylemathias** [comment]
@lmmrssa I was wrong when i said it was going to be 600mb (i updated Leonards comment). The APK is around 60mb .

**Comment by leomaxi** [comment]
@lmmrssa The raw take-home apk is targeted to be Max 60MB. Currently, It embeds Adobe Reader.apk (12.4MB) and Firefox.apk (37.8MB). 
When this issue (https://github.com/open-learning-exchange/take-home/issues/23#) is resolved, we will then make these two apk's external files -: for users to download if not installed.

**Comment by sente** [comment]
I don't think we should be embedding Firefox and Adobe Reader.  It's not unreasonable to require them but seldom do I see an app _embedding_ other apk's.  Why not simply have links to the apks?  Adobe and Firefox make up 50 of the total 60MB.
